src/sentry/hybridcloud/models/webhookpayload.py

```diff
@@ -46,6 +48,23 @@ class Meta:
         indexes = (
             models.Index(fields=["mailbox_name"]),
             models.Index(fields=["schedule_for"]),
+            models.Index(fields=["provider"], name="webhookpayload_provider_idx"),
+            models.Index(
+                fields=["mailbox_name", "id"],
+                name="webhookpayload_mailbox_id_idx",
+            ),
+            models.Index(
+                ExpressionWrapper(
+                    Case(
+                        When(provider="stripe", then=Value(1)),
+                        default=Value(10),
+                        output_field=IntegerField(),
+                    ),
+                    output_field=IntegerField(),
+                ),
+                F("id"),
+                name="webhookpayload_priority_idx",
```

Comment:
If you're also computing priority in the query from #86304 would this index be used?

Reply:
It should be used, as long as the query matches this

Reply:
I think it should be used as well. It's an index on expressions: https://www.postgresql.org/docs/current/indexes-expressional.html

Reply:
Having a look at the query, I don't think `webhookpayload_priority_idx` ended up being useful. I think that because of the 
```
id in (
    select min(id)
    from hybridcloud_webhookpayload
    group by mailbox_name
)
``` subquery, we end up using `webhookpayload_mailbox_id_idx` in the query query, and then just `hybridcloud_webhookpayload_pkey` to filter to the relevant rows.

So we can probably just remove it. I think that `webhookpayload_mailbox_id_idx` was a good addition though.